File: parish360-core/src/main/java/org/parish360/core/configurations/dto/ParishYearInfo.java
package org.parish360.core.configurations.dto;

import jakarta.validation.constraints.NotNull;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

import java.time.LocalDate;
import java.time.LocalDateTime;

@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
public class ParishYearInfo {
    private String id;
    private LocalDateTime createdAt;
    private String createdBy;
    private LocalDateTime updatedAt;
    private String updatedBy;

    @NotNull(message = "name cannot be empty")
    private String name;

    private LocalDate startDate;
    private LocalDate endDate;
    private String status;
    private boolean locked;
    private String comment;
    private String parishId;
}

File: parish360_mobile/lib/features/parish-year/domain/entities/parish_year_info.dart
class ParishYearInfo {
  final String? id;
  final DateTime? createdAt;
  final String? createdBy;
  final DateTime? updatedAt;
  final String? updatedBy;
  final String? name;
  final DateTime? startDate;
  final DateTime? endDate;
  final String? status;
  final bool? locked;
  final String? comment;
  final String? parishId;

  ParishYearInfo({
    this.id,
    this.createdAt,
    this.createdBy,
    this.updatedAt,
    this.updatedBy,
    this.name,
    this.startDate,
    this.endDate,
    this.status,
    this.locked,
    this.comment,
    this.parishId,
  });

  factory ParishYearInfo.fromJson(Map<String, dynamic> json) {
    return ParishYearInfo(
      id: json['id'] as String?,
      createdAt: json['created_at'] != null
          ? DateTime.parse(json['created_at'] as String)
          : null,
      createdBy: json['created_by'] as String?,
      updatedAt: json['updated_at'] != null
          ? DateTime.parse(json['updated_at'] as String)
          : null,
      updatedBy: json['updated_by'] as String?,
      name: json['name'] as String?,
      startDate: json['start_date'] != null
          ? DateTime.parse(json['start_date'] as String)
          : null,
      endDate: json['end_date'] != null
          ? DateTime.parse(json['end_date'] as String)
          : null,
      status: json['status'] as String?,
      locked: json['locked'] as bool?,
      comment: json['comment'] as String?,
      parishId: json['parish_id'] as String?,
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'id': id,
      'created_at': createdAt?.toIso8601String(),
      'created_by': createdBy,
      'updated_at': updatedAt?.toIso8601String(),
      'updated_by': updatedBy,
      'name': name,
      'start_date': startDate?.toIso8601String(),
      'end_date': endDate?.toIso8601String(),
      'status': status,
      'locked': locked,
      'comment': comment,
      'parish_id': parishId,
    };
  }
}
